### PR TITLE
docs: add docs codeowners

### PR DESCRIPTION
**Pull request by harish-sethuraman**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

> The people you choose as code owners must have write permissions for the repository. When the code owner is a team, that team must be visible and it must have write permissions, even if all the individual members of the team already have write permissions directly, through organization membership, or through another team membership.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+/docs/ @eslint/website-team
```
Comment (nzakas) [inline .github/CODEOWNERS:1]: making sure everyone can approve

```suggestion
/docs/ @eslint/website-team @eslint/eslint-team
```

**Review by nzakas — CHANGES_REQUESTED** [review]
we also need to set eslint/eslint-team as the owners for everything outside of docs to limit approvals.

File: .github/CODEOWNERS (hunk 1 above)
Comment (harish-sethuraman) [reply .github/CODEOWNERS]: done

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1 @@
+/docs/ @eslint/website-team @eslint/eslint-team
```
Comment (nzakas) [inline .github/CODEOWNERS:1]:
```suggestion
/docs/ @eslint/website-team @eslint/eslint-team
!/docs/ @eslint/eslint-team
```

**Review by nzakas — APPROVED** [review]
Okay, let's give this a try.

**Pull request merged**